app/src/main/kotlin/com/getstrm/pace/processing_platforms/bigquery/BigQueryClient.kt

```diff
@@ -123,20 +133,23 @@ class BigQueryClient(
 
     override val type = BIGQUERY
 
-    override suspend fun listGroups(): List<Group> {
+    override suspend fun listGroups(pageParameters: PageParameters): PagedCollection<Group> {
         val query = """
             SELECT
               DISTINCT userGroup
             FROM
               $userGroupsTable
+            ORDER BY userGroup
+            LIMIT ${pageParameters.pageSize}
+            OFFSET ${pageParameters.skip}
```

Comment:
Have we tested whether or not we can do this with jooq templating? If possible, we should do it here

Reply:
no I haven't.

Reply:
Jooq provides typesafety. A plain string containing SQL isn't. I agree with the fact that we don't use commercial Jooq (which we can include in PACE now, as PACE is OS, Jooq Commercial can be used freely IIRC), which prevents us from using the correct dialect.